tests/fixtures/full-content.test.ts

```diff
@@ -0,0 +1,283 @@
+	getAvailableBlockFixturesBasenames,
+	getBlockFixtureHTML,
+	getBlockFixtureJSON,
+	getBlockFixtureParsedJSON,
+	getBlockFixtureSerializedHTML,
+	writeBlockFixtureParsedJSON,
+	writeBlockFixtureJSON,
+	writeBlockFixtureSerializedHTML,
+} from './utils';
+import { registerWooBlocks } from './registerBlocks';
+import '../../assets/js/categories/register-woo-category';
+
+const blockBasenames = getAvailableBlockFixturesBasenames();
+
+/**
+ * Returns only the properties of the block that
+ * we care about comparing with the fixture data.
+ *
+ * @param {WPBlock[]} blocks loaded blocks to normalize.
+ */
+const normalizeParsedBlocks = ( blocks ) =>
+	blocks.map( ( block ) => ( {
+		name: block.name,
+		isValid: block.isValid,
+		attributes: JSON.parse( JSON.stringify( block.attributes ) ),
+		innerBlocks: normalizeParsedBlocks( block.innerBlocks ),
+	} ) );
+
+describe( 'full post content fixture', () => {
+	beforeAll( () => {
+		const blockMetadataFiles = glob.sync( 'assets/js/blocks/*/block.json' );
+		const blockDefinitions = Object.fromEntries(
+			blockMetadataFiles.map( ( file ) => {
+				// eslint-disable-next-line @typescript-eslint/no-var-requires
+				const { name, ...metadata } = require( file );
+				return [ name, metadata ];
+			} )
+		);
+		unstable__bootstrapServerSideBlockDefinitions( blockDefinitions );
+		registerWooBlocks();
+	} );
+
+	let spacer = 4;
+	if ( prettierConfig?.useTabs ) {
+		spacer = '\t';
+	} else if ( prettierConfig?.tabWidth ) {
+		spacer = prettierConfig?.tabWidth;
+	}
+
+	blockBasenames.forEach( ( basename ) => {
+		// eslint-disable-next-line jest/valid-title
+		it( basename, () => {
+			const { filename: htmlFixtureFileName, file: htmlFixtureContent } =
+				getBlockFixtureHTML( basename );
+			if ( htmlFixtureContent === null ) {
+				throw new Error(
+					`Missing fixture file: ${ htmlFixtureFileName }`
+				);
+			}
+
+			const {
+				filename: parsedJSONFixtureFileName,
+				file: parsedJSONFixtureContent,
+			} = getBlockFixtureParsedJSON( basename );
+			const parserOutputActual = grammarParse( htmlFixtureContent );
+			let parserOutputExpectedString;
+			if ( parsedJSONFixtureContent ) {
+				parserOutputExpectedString = parsedJSONFixtureContent;
+			} else if ( process.env.GENERATE_MISSING_FIXTURES ) {
+				parserOutputExpectedString =
+					JSON.stringify( parserOutputActual, null, spacer ) + '\n';
+				writeBlockFixtureParsedJSON(
+					basename,
+					parserOutputExpectedString
+				);
+			} else {
+				throw new Error(
+					`Missing fixture file: ${ parsedJSONFixtureFileName }`
+				);
+			}
+
+			const parserOutputExpected = JSON.parse(
+				parserOutputExpectedString
+			);
+			try {
+				expect( parserOutputActual ).toEqual( parserOutputExpected );
+			} catch ( err ) {
+				throw new Error(
+					format(
+						"File '%s' does not match expected value:\n\n%s",
+						parsedJSONFixtureFileName,
+						err.message
+					)
+				);
+			}
+
+			const blocksActual = parse( htmlFixtureContent );
```

Comment:
An issue at the moment here is that `parse()` here returns `[]` because [it can't find the registered block in the store](https://github.com/WordPress/gutenberg/blob/trunk/packages/blocks/src/api/parser/index.js#L204).